cis194/week3/zehua/src/Exercises.hs

```diff
@@ -0,0 +1,28 @@
+module Exercises where
+
+-- import           Data.List (dropWhileEnd)
+
+-- ex1
+
+-- helper function adapted from https://stackoverflow.com/a/2028218
+-- doing `drop n xs` instead of `drop (n-1) xs` to save 4 chars :)
+--   which requires passing in 0-based `zip [0..]` instead of `zip [1..]`
+-- keeping the name `every` instead of a short one-character name, e.g., `e`
+every :: (Int, [a]) -> [a]
+every (n, xs) = case drop n xs of
+              (y:ys) -> y : every (n, ys)
+              []     -> []
+
+skips :: [a] -> [[a]]
+-- infinite list first then cut off
+-- skips = takeWhile (not . null) . map every . zip [0..] . repeat
+-- dropWhileEnd does not seem to work on infinite list :|
+-- skips = dropWhileEnd null . map every . zip [0..] . repeat
+
+-- <*> S combinator on reader monad to implement s x y z = (x z) (y z)
+--   https://kseo.github.io/posts/2016-12-24-reader-monad-and-ski-combinators.html
+-- skips = map every . zip [0..] . (flip replicate <*> length)
```

Comment:
@shineli1984 is there a way to get rid of the bracket?

Reply:
`. ap replicate’ $ length where replicate’ = ...`